constructor/osx/update_path.sh

```diff
@@ -8,4 +8,35 @@ set -eux
 
 PREFIX="$2/__NAME_LOWER__"
 PREFIX=$(cd "$PREFIX"; pwd)
-"$PREFIX/bin/python" -m conda init --all
+INIT_FILES=$("$PREFIX/bin/python" -m conda init --all | tee)
+
+# Just like in run_install.sh, the files generated by the installer
+# are owned by root when installed outside of $HOME. So, ownership of
+# files modified by conda init must be changed to belong to $USER.
+# For shells like fish or powershell, conda init creates subdirectories
+# inside $HOME and/or $PREFIX, so ensure that parent directories also
+# have the correct owner.
+if [[ "${USER}" != "root" ]]; then
+    echo "Fixing permissions..."
+    read -r -a MODIFIED_FILES <<< "$(\
+      echo "${INIT_FILES}" |\
+      grep modified |\
+      awk '{print $2}' |\
```

Comment:
```suggestion
      awk '/modified/{print $2}' |\
```

I think this is equivalent. You can probably add the `grep` call below here as well, in between the slashes.